hub/api/tensor.py

```diff
@@ -178,7 +178,7 @@ def __len__(self):
             int: The current length of this tensor.
         """
 
-        return self.index.length(self.chunk_engine.num_samples)
+        return self.index.length(self.meta.length)
```

Comment:
can we add a check here to ensure both of these are equal so that if they ever get out of sync it's easy to catch?

Reply:
like an exception?

Reply:
yeah, an exception that says something like "the tensor meta and chunk engine are out of sync"

Reply:
on it